File: mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // initialize blocks and gaps for pyramid
    string block = "#";
    string gap = " ";
    // prompt user for numbers between 1 and 8 inclusive, continue loop until user follows prompt
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (!(n >= 1 && n <= 8));
    // create loop for printing blocks and gaps
    for (int i = 0; i < n; i++)
    {
        for (int j = n - 1; j > i; j--)
        {
            printf("%s", gap);
        }
        for (int k = i + 1; k > 0; k--)
        {
            printf("%s", block);
        }
        printf("\n");
    }
}

File: mario.py
# TODO
from cs50 import get_int


# Creating block and gap variables
block = "#"
gap = " "


def main():
    # Prompt user for height of pyramid until input is properly received
    while True:
        size = get_int("Height: ")
        if size >= 1 and size <= 8:
            break
    # Loops which iteratively print blocks and gaps
    i = 0
    for i in range(size):
        for j in range(size - i - 1):
            print(gap, end="")
        for k in range(i + 1):
            print(block, end="")
        print()


main()
